packages/scripts/scripts/check-licenses.js

```diff
@@ -258,6 +258,7 @@ modules.forEach( ( path ) => {
 					}, stringDetectedType );
 				}, detectedType );
 			}
+			return false;
```

Comment:
I think we should not alter whatever the current reduced value is, no?

```suggestion
			return detectedType;
```

The logic is hard to follow, but I wonder also if this shouldn't be implemented as an `Array#find`. That might be a bit more of a refactor than we want to do here, though.